src/main/java/emu/grasscutter/command/commands/GiveAllCommand.java

```diff
@@ -0,0 +1,109 @@
+import emu.grasscutter.game.props.ActionReason;
+import emu.grasscutter.server.packet.send.PacketItemAddHintNotify;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@Command(label = "giveall", usage = "giveall [player] <amount>",
+        description = "Gives All item to you or the specified player", aliases = {"givea"}, permission = "player.giveall")
+public class GiveAllCommand implements CommandHandler {
+
+    @Override
+    public void execute(GenshinPlayer sender, List<String> args) {
+        int target,amount=99999;
+
+        switch (args.size()) {
+            default: // giveall *no args*
+                target = sender.getUid();
+                break;
+            case 1: //[player]
+                try {
+                    target = Integer.parseInt(args.get(0));
+                    if (Grasscutter.getGameServer().getPlayerByUid(target) == null) {
+                        CommandHandler.sendMessage(sender, "Invalid player ID.");
+                        return;
+                    }
+                }catch (NumberFormatException ignored){
+                    CommandHandler.sendMessage(sender, "Invalid amount or player ID.");
+                    return;
+                }
+                break;
+            case 2: //[player] [amount]
+                try {
+                    target = Integer.parseInt(args.get(0));
+                    if (Grasscutter.getGameServer().getPlayerByUid(target) == null && sender != null) {
+                        target = sender.getUid();
+                        amount = Integer.parseInt(args.get(0));
+                    } else {
+                        amount = Integer.parseInt(args.get(1));
+                    }
+                 } catch (NumberFormatException ignored) {
+                    CommandHandler.sendMessage(sender, "Invalid amount or player ID.");
+                    return;
+                }
+                break;
+        }
+        GenshinPlayer targetPlayer = Grasscutter.getGameServer().getPlayerByUid(target);
+
+        if (targetPlayer == null) {
+            CommandHandler.sendMessage(sender, "Player not found.");
+            return;
+        }
+
+        this.GetAllItem(targetPlayer,amount);
+        CommandHandler.sendMessage(sender, String.format("Get All Items Done."));
+    }
+
+    public void GetAllItem(GenshinPlayer player, int amount){
+        CommandHandler.sendMessage(player, "Get All Items...");
+        for (ItemData itemdata: GenshinData.getItemDataMap().values()) {
+            if(itemdata.getId() > 1000 && itemdata.getId() <= 2000)continue;//is avatar
+            if (itemdata.isEquip()) {
+                List<GenshinItem> items = new LinkedList<>();
+                for (int i = 0; i < 20; i++) {
+                    items.add(new GenshinItem(itemdata));
+                }
+                player.getInventory().addItems(items);
+                player.sendPacket(new PacketItemAddHintNotify(items, ActionReason.SubfieldDrop));
```

Comment:
Recommended to remove preview packet so server doesn’t overflow with packets

Reply:
ok i will solve it